spring-kafka/src/main/java/org/springframework/kafka/core/KStreamBuilderFactoryBean.java

```diff
@@ -116,8 +130,12 @@ public synchronized void start() {
 		if (!this.running) {
 			try {
 				this.kafkaStreams = new KafkaStreams(getObject(), this.streamsConfig, this.clientSupplier);
-				this.kafkaStreams.setStateListener(this.stateListener);
-				this.kafkaStreams.setUncaughtExceptionHandler(this.exceptionHandler);
+				if (this.stateListener != null) {
```

Comment:
I didn't do that knowingly because `KafkaStreams` lets to have and configure them as null.

Although I won't mind if you insist to leave your fix.